ecoembes/models/product_template.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    ecoembes_product_master = fields.Char(
+        string="Product master",
+        compute="_compute_ecoembes_info",
+        inverse="_inverse_ecoembes_info",
+    )
+    ecoembes_date_expiry = fields.Date(
+        string="Date expiry",
+        compute="_compute_ecoembes_info",
+        inverse="_inverse_ecoembes_info",
+    )
+    ecoembes_sig = fields.Boolean(
+        string="SIG/NOSIG option",
+        compute="_compute_ecoembes_info",
+        inverse="_inverse_ecoembes_info",
+    )
+    ecoembes_comment = fields.Text(
+        string="Comments",
+        compute="_compute_ecoembes_info",
+        inverse="_inverse_ecoembes_info",
+    )
+    weight_net_gram_round = fields.Integer(
+        string="Weight (g)",
+        compute="_compute_ecoembes_info",
+        inverse="_inverse_ecoembes_info",
+    )
+    composition_ids = fields.One2many(
+        comodel_name="product.composition",
+        inverse_name="product_tmpl_id",
+        string="Compositions",
+    )
+
+    @api.depends(
+        "product_variant_ids",
+        "product_variant_ids.ecoembes_product_master",
+        "product_variant_ids.ecoembes_date_expiry",
+        "product_variant_ids.ecoembes_sig",
+        "product_variant_ids.ecoembes_comment",
+        "product_variant_ids.weight_net_gram_round",
+    )
+    def _compute_ecoembes_info(self):
+        unique_variants = self.filtered(lambda x: len(x.product_variant_ids) == 1)
+        for item in unique_variants:
+            variant = item.product_variant_ids
+            item.ecoembes_product_master = variant.ecoembes_product_master
+            item.ecoembes_date_expiry = variant.ecoembes_date_expiry
+            item.ecoembes_sig = variant.ecoembes_sig
+            item.ecoembes_comment = variant.ecoembes_comment
+            item.weight_net_gram_round = variant.weight_net_gram_round
+        for item in self - unique_variants:
```

Comment:
Instead of iterating this way, just assign at the beginning at once to all records `False`.